src/sidecar/middlewares.ts

```diff
@@ -113,3 +114,46 @@ export class ErrorResponseMiddleware implements Middleware {
     }
   }
 }
+
+/** Track the number of requests to Confluent Cloud that have happened recently, controlled by
+ * {@link CCloudRecentRequestsMiddleware}. */
+export let numRecentCCloudRequests: number = 0;
+/** Middleware to track the number of requests to Confluent Cloud that have happened recently. */
+export class CCloudRecentRequestsMiddleware implements Middleware {
+  async pre(context: RequestContext): Promise<void> {
+    // increment the count of pending requests to Confluent Cloud
+    if (hasCCloudConnectionIdHeader(context.init.headers)) {
+      const origValue: number = numRecentCCloudRequests;
+      numRecentCCloudRequests++;
+      logger.debug(`CCloud requests pending: ${origValue} -> ${numRecentCCloudRequests}`);
+    }
+  }
+  async post(context: ResponseContext): Promise<void> {
+    // decrement the count of pending requests to Confluent Cloud after a delay so we get a feeling
+    // for "recent activity" instead of the exact current state of pending requests (which will
+    // likely be 0 unless someone get the timing just right)
+    if (hasCCloudConnectionIdHeader(context.init.headers)) {
+      const origValue: number = numRecentCCloudRequests;
+      setTimeout(() => {
+        numRecentCCloudRequests--;
+        logger.debug(
+          `CCloud requests pending (delayed): ${origValue} -> ${numRecentCCloudRequests}`,
+        );
+      }, 15_000);
```

Comment:
Any particular reason for 15 seconds?

Reply:
No specific reason other than it seemed like a reasonable amount of time -- do you think it should be shorter/longer?